ground/src/main/java/com/google/android/ground/ui/datacollection/components/LoiNameDialog.kt

```diff
@@ -0,0 +1,129 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection.components
+
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.ButtonColors
+import androidx.compose.material3.OutlinedButton
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.material3.TextField
+import androidx.compose.material3.TextFieldColors
+import androidx.compose.material3.TextFieldDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.em
+import androidx.core.content.ContextCompat
+import com.google.android.ground.R
+import com.google.android.ground.ui.common.AbstractFragment
+import com.google.android.material.color.MaterialColors
+
+private fun AbstractFragment.getColor(id: Int): Color =
+  Color(ContextCompat.getColor(requireContext(), id))
+
+private fun AbstractFragment.getMaterialColor(id: Int): Color =
+  Color(MaterialColors.getColor(requireContext(), id, ""))
+
+@Composable
+private fun AbstractFragment.getElementColors(): Triple<ButtonColors, Color, TextFieldColors> {
+  val primaryColor = getMaterialColor(R.attr.colorPrimary)
+  val onPrimaryColor = getColor(R.color.md_theme_onPrimary)
+  val onSurfaceDisabledColor = getColor(R.color.md_theme_on_surface_disabled)
+  val textFieldColor = getColor(R.color.md_theme_textFieldContainers)
+  val saveButtonColors =
+    ButtonColors(
+      containerColor = primaryColor,
+      contentColor = onPrimaryColor,
+      disabledContainerColor = onSurfaceDisabledColor,
+      disabledContentColor = onPrimaryColor,
+    )
+  val cancelButtonColor = getMaterialColor(R.attr.colorPrimary)
+  val textFieldColors =
+    TextFieldDefaults.colors(
+      focusedIndicatorColor = primaryColor,
+      unfocusedIndicatorColor = primaryColor,
+      focusedContainerColor = textFieldColor,
+      unfocusedContainerColor = textFieldColor,
+      cursorColor = primaryColor,
+    )
+
+  return Triple(saveButtonColors, cancelButtonColor, textFieldColors)
+}
+
+@Composable
+fun LoiNameDialog(
+  fragment: AbstractFragment,
```

Comment:
Doesn't passing a fragment reference, make this function dependent?
How about we use the State-Hoisting approach and let all the values come from a specific state, no matter which fragment or activity or anyone call this composable function.

WDYT?

Reply:
Ah, the only reason I was sending in the fragment was for its context... which I now see is not necessary with Composable's `colorResource()`. I was able to find a solution to replace the color attribute as well.
https://stackoverflow.com/questions/62971113/how-to-reference-theme-attributes-in-jetpack-compose

Removed reference to `fragment`.